iree/compiler/Dialect/HAL/Target/LLVM/LinkerTool.cpp

```diff
@@ -159,14 +159,22 @@ static std::string findToolAtPath(
   return "";
 }
 
-LogicalResult LinkerTool::runLinkCommand(const std::string &commandLine) {
-  LLVM_DEBUG(llvm::dbgs() << "Running linker command:\n" << commandLine);
-  auto escapedCommandLine = escapeCommandLineComponent(commandLine);
-  int exitCode = system(escapedCommandLine.c_str());
+LogicalResult LinkerTool::runLinkCommand(std::string commandLine,
+                                         StringRef env) {
+  LLVM_DEBUG(llvm::dbgs() << "Running linker command:\n"
+                          << env << " " << commandLine);
+  if (!env.empty()) {
+#if defined(_MSC_VER)
+    commandLine = ("set " + env + " && " + commandLine).str();
+#else
+    commandLine = (env + " " + commandLine).str();
+#endif  // _MSC_VER
+  }
+  int exitCode = system(commandLine.c_str());
   if (exitCode == 0) return success();
   llvm::errs() << "Linking failed; escaped command line returned exit code "
                << exitCode << ":\n\n"
-               << escapedCommandLine << "\n\n";
+               << commandLine << "\n\n";
   return failure();
 }
```

Comment:
Just noticed after the fact, but this broke the system linker on Windows (untested since the embedded linker is the default).

The `auto escapedCommandLine = escapeCommandLineComponent(commandLine);` was load bearing.

I'm including a fix in https://github.com/google/iree/pull/7810.